pinecone/test_suite.go

```diff
@@ -126,27 +126,26 @@ func createCollection(ts *IntegrationTests, ctx context.Context) {
 }
 
 func WaitUntilIndexReady(ts *IntegrationTests, ctx context.Context) (bool, error) {
-	maxRetries := 24
+	start := time.Now()
 	delay := 5 * time.Second
-	totalSeconds := 0
+	maxWaitTimeSeconds := 280 * time.Second
 
-	for i := 0; i < maxRetries; i++ {
+	for {
```

Comment:
After much debugging, I'm fairly sure there was an issue in how this was polling previously. Tweaked a bit to hopefully be more straightforward.

Reply:
oh no! apologies!